Example of how to setup a sidecar app

johnman opened the pull request.

johnman committed:
Example of how to setup a sidecar app

obany commented on how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts:
```diff
@@ -0,0 +1,217 @@
+ */
+async function initializeDom(): Promise<void> {
+	loggingElement = document.querySelector("#logging");
+	const loggingContainer: HTMLDivElement | null = document.querySelector("#logging-container");
```
Just a comment really if you use generic `document.querySelector<HTMLDivElement>()` the loggingContainer is already the correct type

johnman replied:
done

obany commented on how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts:
```diff
@@ -0,0 +1,217 @@
+ * @param error The error to inspect and retrieve the message from
+ * @returns The error message
+ */
+function getErrorMessage(error: unknown): string {
```
We have a sort of standardized method like this, which we use all over the place https://github.com/built-on-openfin/workspace-starter/blob/bcf97bdf9b4cd8ca32ebed835b3483db6ffdef5d/how-to/workspace-platform-starter/client/src/framework/utils.ts#L112-L119

johnman replied:
copied (didn't copy the util file as I am trying to keep it small but copied the function)

obany commented on how-to.v2/use-a-sidecar-app-basic/platform/client/tsconfig.json:
```diff
@@ -0,0 +1,16 @@
+		"moduleResolution": "node",
+		"types": ["./fin"]
+	},
+	"include": ["./src/**/*.ts", "fin.d.ts"]
```
not sure why fin.d.ts is in here, if anything the types on line 13 should say ["./types"] and the fin.d.ts file should be in a types folder, to match other repos

johnman replied:
updated

obany commented on how-to.v2/use-a-sidecar-app-basic/platform/public/js/provider.bundle.js:
```diff
@@ -0,0 +1,228 @@
+/******/ (() => { // webpackBootstrap
```
Should this file be excluded in the gitignore

johnman replied:
added to gitignore

obany commented on how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts:
```diff
@@ -0,0 +1,217 @@
+ * @returns channel client
+ */
+async function createChannelClient(): Promise<ChannelClient> {
+	const channelName = "sidecar-app";
```
Can we make the channel names global consts and the same in the C# app, easier to spot them

johnman replied:
done

obany commented on how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts:
```diff
@@ -0,0 +1,217 @@
+		});
+
+		// received request over channel to display a message
+		channelClient.register('sidecar-app-client-subscriber', async (payload) => {
```
Same with naming here, can we make global const

johnman replied:
done

obany commented on how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts:
```diff
@@ -0,0 +1,217 @@
+	try {
+		if(channelClient !== undefined) {
+			loggingAddEntry("Sending Message to SideCar App.");
+			let response = await channelClient.dispatch("sidecar-app-echo", "Message from platform to sidecar");
```
Global const

johnman replied:
done

obany commented on how-to.v2/use-a-sidecar-app-basic/sidecar-app/Provider.cs:
```diff
@@ -0,0 +1,109 @@
+            onAction("Creating Provider");
+            provider = channels.CreateProvider("sidecar-app");
+            onAction("Registering echo topic.");
+            provider.RegisterTopic<string, string>("sidecar-app-echo", (payload) =>
```
Move the topics to global consts

johnman replied:
done

obany commented on how-to.v2/use-a-sidecar-app-basic/sidecar-app/Provider.cs:
```diff
@@ -0,0 +1,109 @@
+        {
+            if(provider != null)
+            {
+                await provider.BroadcastAsync("sidecar-app-client-subscriber", message);
```
Global const

johnman replied:
done

obany reviewed (approved):
Approved, mentioned a few things but nothing that stops it working :)

johnman committed:
Applied feedback and added a few tweaks

Overwrite was needed for the zip step.
Logging flows in the same way in SideCarApp and SideCarPlatform.
